.github/workflows/reports.yml

```diff
@@ -316,28 +336,57 @@ jobs:
           mv /home/runner/work/noir/noir/scripts/test_programs/compilation_report.sh ./compilation_report.sh
           chmod +x ./compilation_report.sh
           ./compilation_report.sh 1
+      
+      - name: Generate execution report
+        working-directory: ./test-repo/${{ matrix.project.path }}
+        run: |
+          mv /home/runner/work/noir/noir/scripts/test_programs/execution_report.sh ./execution_report.sh
+          IS_LIBRARY=${{ matrix.project.is_library }}
+          if [[ "$IS_LIBRARY" != "true" ]]; then
+            ./execution_report.sh 1
+          fi
 
       - name: Move compilation report
-        id: report
+        id: compilation_report
         shell: bash
         run: |
           PACKAGE_NAME=${{ matrix.project.path }}
           PACKAGE_NAME=$(basename $PACKAGE_NAME)
           mv ./test-repo/${{ matrix.project.path }}/compilation_report.json ./compilation_report_$PACKAGE_NAME.json
           echo "compilation_report_name=$PACKAGE_NAME" >> $GITHUB_OUTPUT
 
+      - name: Move execution report
+        id: execution_report
+        shell: bash
+        run: |
+          IS_LIBRARY=${{ matrix.project.is_library }}
+          if [[ "$IS_LIBRARY" != "true" ]]; then
```

Comment:
You can add a `if: ${{ !matrix.project.is_library }}` to the job directly (along with any later steps). I think this is cleaner.